pgmpy/prediction/DynamicDMLRegressor.py

```diff
@@ -0,0 +1,810 @@
+#!/usr/bin/env python3
+"""
+Dynamic Double/Debiased Machine Learning for Sequential Treatment Effects.
+"""
+
+import warnings
+
+import numpy as np
+from scipy.stats import norm
+from sklearn.base import BaseEstimator, RegressorMixin, clone
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.linear_model import LinearRegression
+from sklearn.model_selection import GroupKFold
+from sklearn.utils.validation import check_array, check_is_fitted
+
+
+class DynamicDMLRegressor(RegressorMixin, BaseEstimator):
+    """
+    Dynamic Double/Debiased Machine Learning for sequential treatment effects.
+    Estimates causal effects of treatments at different time periods on a
+    final outcome using recursive g-estimation with Neyman orthogonal moments.
+
+    Parameters
+    ----------
+    causal_graph : DAG, PDAG, ADMG, MAG, or PAG, optional
+        Causal graph with defined variable roles. If provided, the graph must have
+        the following roles: `exposure`, `outcome`. Additionally, `adjustment` and
+        `pretreatment` can be specified for confounders.
+        If None, all features in X are treated as confounders, and T/y column names
+        are used directly.
+
+    model_y : estimator or 'auto', default='auto'
+        Machine learning model for outcome E[Y | X]. Must implement fit() and predict().
+        If 'auto': LinearRegression for continuous outcomes, RandomForestRegressor for discrete.
+
+    model_t : estimator or 'auto', default='auto'
+        Machine learning model for treatment E[T | X]. Must implement fit() and predict().
+        If 'auto': LinearRegression for continuous treatments, RandomForestClassifier for discrete.
+
+    cv : int, default=2
+        Number of cross-fitting folds (minimum 2). Internally uses GroupKFold to respect
+        panel structure when groups parameter is provided.
+
+    n_periods : int or None, default=None
+        Number of treatment periods. If None, inferred from unique values in groups.
+
+    discrete_treatment : bool, default=False
```

Comment:
do we not know this, based on the type of the column? We could have a `None` = automatic setting where the estimator tries to infer this.